targets/teensy40.json

```diff
@@ -0,0 +1,25 @@
+{
+  "inherits": ["cortex-m"],
+  "llvm-target": "armv7em-none-eabi",
+  "cpu": "cortex-m7",
+  "build-tags": ["teensy40", "teensy", "mimxrt1062", "nxp"],
+  "automatic-stack-size": false,
+  "default-stack-size": 131072,
```

Comment:
Isn't 128K a bit on the large side? I would suggest 2K or 4K instead (in fact, most chips use 1K and that's usually enough).

Reply:
Definitely on the large side! Wasn't sure if that would be problematic or not. 

The reason for this, is that an entire 128K physical RAM bank (whose accesses are all performed via the 600 MHz AHB bus) is completely unused in TinyGo! This seemed like an unfortunate waste of resources, so I just handed it all over for the stack. The heap is allocated on the other physical 128K RAM bank.

(I need a guru such as yourself to help allocate code to this unused bank during startup so that we have super fast code accesses. Currently it is all stored in flash, and we rely on the MPU (which is configured in this PR!) to cache these for fast code accesses from flash)

Reply:
Changed to 4K (4096)

Reply:
This is the stack size for goroutines, which are allocated on the heap. The scheduler has a different stack which is set in the linker script (`_stack_size`).